typescript/sdk/src/token/deploy.ts

```diff
@@ -135,11 +135,30 @@ export class HypERC20Deployer extends GasRouterDeployer<
     chain: ChainName,
     config: HypERC20CollateralConfig,
   ): Promise<HypERC20Collateral> {
-    return this.deployContract(
-      chain,
-      isFastConfig(config) ? TokenType.fastCollateral : TokenType.collateral,
-      [config.token, config.mailbox],
-    );
+    let contractName:
+      | TokenType.fastCollateral
+      | TokenType.collateral
+      | TokenType.collateralVault;
+    let constructorArgs: Parameters<
+      HypERC20Factories[typeof contractName]['deploy']
+    >;
+    switch (config.type) {
+      case TokenType.fastSynthetic || TokenType.fastCollateral:
+        contractName = TokenType.fastCollateral;
+        constructorArgs = [config.token, config.mailbox];
+        break;
+      case TokenType.collateral:
+        contractName = TokenType.collateral;
+        constructorArgs = [config.token, config.mailbox];
+        break;
+      case TokenType.collateralVault:
+        contractName = TokenType.collateralVault;
+        constructorArgs = [config.vaultAddress, config.mailbox];
```

Comment:
ditto on this, you can reuse token config and skip this

Reply:
Fixed